ryo-modal.el

```diff
@@ -264,5 +264,74 @@ See `ryo-modal-keys' for more information."
       (set-cursor-color ryo-modal-cursor-color)
     (set-cursor-color ryo-modal-default-cursor-color)))
 
+(defun ryo-modal--extract-commands-from (args)
+  "Extract commands from arglist to enable lazy loading for :ryo-modal."
+  (let (commands)
+    (while args
+      (cond
+       ((listp (car (cdar args)))
+        (setq commands (append commands (ryo-modal--extract-commands-from (car (cdar args))))))
+       ((equal (car (cdar args)) :hydra)
+        (setq my-hydra-args (car (cdr (cdar args))))
```

Comment:
`my-hydra-args` seems to be used locally, so we should use `let` since `setq` will clutter the global namespace.